### PR TITLE
Updated the architecture document including the diagram to include the reverse vpn tunnel and sni.

### LINKED ISSUE #4120: Update architecture documentation
**How to categorize this issue?**

/area documentation
/kind enhancement
/priority 4

**What would you like to be added**:
Update the architecture picture (https://github.com/gardener/gardener/blob/master/docs/concepts/architecture.md) and description to reflect changes with regards to "reversed cluster VPN" and "shoot API server via SNI"

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
It updates the architecture document and adds an updated version of the architecture diagram. It is based on https://github.com/gardener/documentation/pull/248.

**Which issue(s) this PR fixes**:
Fixes #4120 

**Special notes for your reviewer**:

**Release note**:

```other operator
None.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Thanks! In the picture, there is an "request arrow" from the `Istio Ingress LB` to the `API Server` of the shoot, but doesn't it first go to `Envoy` and only from their to the `API Server`?

**Comment by ScheererJ** [comment]
> Thanks! In the picture, there is an "request arrow" from the `Istio Ingress LB` to the `API Server` of the shoot, but doesn't it first go to `Envoy` and only from their to the `API Server`?

You are correct that on a lower level the flow is:
`infrastructure load balancer` => `isto-ingress/envoy` => `kube-apiserver`

I intentionally combined the load balancer and the istio-ingress/envoy components semantically to have a more concise structure. Do you want me to add it?

**Comment by rfranzke** [comment]
OK, no strong opinion, maybe it's out of scope of this PR as the goal was to reflect the changes related to the `ReversedVPN`.

**Review by rfranzke — DISMISSED** [review]
/lgtm

**Comment by rfranzke** [comment]
If there are no other thoughts on this PR, let's merge it by end of today.

**Comment by vpnachev** [comment]
Has the ReservedVPN graduated to beta?
There are other changes in the architecture that can be contributed, like the CSI controllers, rename calico to something more general as cilium is also supported. Gardenlet is not connected only to the Gardener API, but also the Seed and Shoot APIs. There is dependency watchdog, gardener resource manager, etcd druid. Should n't this change also show the ManagedIstio components? What about the apiserver proxy component in the shoots?

**Comment by rfranzke** [comment]
@vpnachev Isn't this out of scope of this PR? The purpose of @ScheererJ was to fix #4120 which he did.
Personally, I think we might want to overhaul the architecture picture to make it less "complex" (I have this on my "TGIF" list, still, didn't make it yet).

**Comment by vpnachev** [comment]
I was just wondering why we want to touch only the VPN that is still [alpha](https://github.com/gardener/gardener/blob/893f4042fa3e4a1814ab04b4dd79e9a09e1f914a/pkg/features/features.go#L94-L97). And instead of depicting the exact implementation, shouldn't we just show that there is some VPN solution that bridges the seed and shoot networks. Currently we have 3 different solutions and obviously we cannot show all of them in this picture.

The rest is for sure out of the scope of this PR. I just wanted to show that there is a lot of components that are implemented but missing in this picture.

File: docs/concepts/architecture.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
...ts inside Kubernetes, is a popular topic in the community, we apply a special pattern catering to the needs of our cloud platform to provision hundreds or even thousands of clusters. We take a so-called "seed" cluster and seed the control plane (such as the API server, scheduler, controllers, etcd persistence and others) of an end-user cluster, which we call "shoot" cluster, as pods into the "seed" cluster. That means one "seed" cluster, of which we will have at least one per IaaS and region, hosts the control planes of multiple "shoot" clusters. That allows us to avoid dedicated hardware/virtual machines for the "shoot" cluster control planes. We simply put the control plane into pods/containers and since the "seed" cluster watches them, they can be deployed with a replica count of 1 and only need to be scaled out when the control plane gets under pressure, but no longer for HA reasons. At the same time, the deployments get simpler (standard Kubernetes deployment) and easier to update (standard Kubernetes rolling update). The actual "shoot" cluster consists only out of the worker nodes (no control plane) and therefore the users may get full administrative access to their clusters.
```
Comment (BeckerMax) [inline docs/concepts/architecture.md:9]: It is hard for me to see the diff here. Can we use a new line after each sentence (formatting output does not change).

Comment (ScheererJ) [reply docs/concepts/architecture.md:9]: The diff is `have one per IaaS` vs.`have at least one per IaaS`.

Do you want me to still split it up into multiple lines?

**Comment by rfranzke** [comment]
@vpnachev What's your concrete suggestion then, holding the PR until the `ReversedVPN` is the default?

**Comment by vpnachev** [comment]
Yes, this is my personal opinion.
But I am fine if rest of the team wants and merge this even now when it is still alpha.

**Comment by rfranzke** [comment]
> Yes, this is my personal opinion.
> But I am fine if rest of the team wants and merge this even now when it is still alpha.

There was no other feedback from anybody else yet. Personally, I'm fine with both. @ScheererJ Would you be okay with closing this PR until the `ReversedVPN` gets promoted to beta, or do you prefer to merge this even now?

**Comment by ScheererJ** [comment]
I did the pull request in response to #4120. No hard feelings if you think that now is not the right time.

@RaphaelVogel What is you opinion as you opened the issue?

**Comment by RaphaelVogel** [comment]
Sorry for jumping in late. The reason for the request to update the picture/documentation is that we are going to start a Beta program on our LIVE landscape for CCloud soon. I was told, that before starting a Beta program, the documentation should be up-to-date. @hoeltcl is this correct?

**Comment by rfranzke** [comment]
@RaphaelVogel In this repository we are maintaining open-source documentation which is/should be independent of internal timelines or milestones of dedicated companies.

**Comment by hoeltcl** [comment]
> Sorry for jumping in late. The reason for the request to update the picture/documentation is that we are going to start a Beta program on our LIVE landscape for CCloud soon. I was told, that before starting a Beta program, the documentation should be up-to-date. @hoeltcl is this correct?

Requirement came from the TM workshop last week. Isn't there an internal location for the architecture diagram?

**Comment by rfranzke** [comment]
Not sure what TM is, but I guess all of this speaks for closing this PR for now and wait for the beta promotion of `ReversedVPN`. Any objections?

**Comment by ScheererJ** [comment]
TM=Threat Modelling

However, the diagram to be updated based on the thread modelling was already updated in #4159.

**Comment by rfranzke** [comment]
Also, see https://github.com/gardener/gardener/pull/4157#issuecomment-859310002, we might need to re-draw this entire picture soon anyways, hopefully less complex and more clean.

**Pull request closed without merging**